### PR TITLE
Add update (PATCH) authorised systems endpoint

### PR DESCRIPTION
**Pull request by Cruikshanks**
In [V1 of the API](https://github.com/charging-module-api) there was an endpoint that allowed the admin user to change the details of an 'authorised system' (our name for user accounts!)

We've always had the intention of bringing it into V2, it was just very low on the list of priorities. Now it's been identified as a need by our test team to allow them to automate the testing of authentication and authorisation.

So, this change adds a new update authorised system endpoint which will allow us to

- amend the status of an authorised system
- amend the name of an authorised system
- amend what regimes an authorised system has access to

### CODE REVIEW

File: app/services/update_authorised_system.service.js [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+      throw Boom.badData('The payload was empty.')
+    }
+    this._validatePayloadStatus(payload)
+    this._validatePayloadName(payload)
+    await this._validatePayloadRegimes(payload)
+  }
+
+  static _validatePayloadStatus (payload) {
+    if (payload.status && !['active', 'inactive'].includes(payload.status)) {
+      throw Boom.badData(`${payload.status} is not valid a status. Can only be active or inactive.`)
```
Comment (StuAA78) [inline app/services/update_authorised_system.service.js:117]:
```suggestion
      throw Boom.badData(`${payload.status} is not a valid status. Can only be active or inactive.`)
```

**Review by StuAA78 — CHANGES_REQUESTED** [review]
Looks good! Just one tiny typo to fix in the validation response.

File: test/services/update_authorised_system.service.test.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+      })
+    })
+
+    describe('but the payload contains an invalid status', () => {
+      it('throws an error', async () => {
+        const payload = { status: 'Happy' }
+        const id = updateAuthSystem.id
+        const err = await expect(UpdateAuthorisedSystemService.go(id, payload))
+          .to
+          .reject(Error, `${payload.status} is not valid a status. Can only be active or inactive.`)
```
Comment (StuAA78) [inline test/services/update_authorised_system.service.test.js:96]:
```suggestion
          .reject(Error, `${payload.status} is not a valid status. Can only be active or inactive.`)
```

File: app/services/update_authorised_system.service.js (hunk 1 above)
Comment (Cruikshanks) [reply app/services/update_authorised_system.service.js:117]: > Typo it is not

Actually, it really is!

**Review by StuAA78 — APPROVED** [review]

**Pull request merged**